File: data/data.code.javascript.client/AckScenePlayers.js
module.exports = AckScenePlayers;

var Packet = require('../net/Packet');
var MsgScenePlayer = require('./MsgScenePlayer');


function AckScenePlayers() {
	// 玩家列表
	this._players = new Array();


	this.Decode = function(packet) {
		var players_count = packet.ReadUshort();
		for (var i = 0; i < players_count; i++)
		{
			 var xx = new MsgScenePlayer();
			 xx.Decode(packet);
			this._players.push(xx);
		}
	}
}

AckScenePlayers.prototype = {
	set Players(val) {
		this._players = val;
	},
	get Players() {
		return this._players;
	},
}

File: data/data.code.lua.client/AckScenePlayers.lua
module("AckScenePlayers", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckScenePlayers})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.players = {}
end


function decode(self, pack)
	local players_count = pack:readU16()
	for i = 1, players_count do
		table.insert(self.players, MsgScenePlayer.new():decode(pack))
	end
	return self
end

return AckScenePlayers
